app/models/user.rb

```diff
@@ -116,6 +116,20 @@ def send_devise_notification(notification, *args)
 		message = devise_mailer.delay.send(notification, self, *args)
 	end
 
+	# override devise class method send_reset_password_instructions to limit to 1 request / 5 min
+	def self.send_reset_password_instructions(attributes={})
+		recoverable = find_or_initialize_with_errors(reset_password_keys, attributes, :not_found)
+		if recoverable.persisted?
+			if recoverable.reset_password_sent_at.nil? || Time.now > recoverable.reset_password_sent_at + 5.minutes
+        recoverable.send_reset_password_instructions
+				return recoverable
+			else
+				recoverable.errors.add(:user, "can't reset password because a request was just sent")
```

Comment:
After thinking some more, this line isn't quite right. If you look at the documentation for [`ActiveModel::Errors#add`](https://api.rubyonrails.org/v4.2/classes/ActiveModel/Errors.html#method-i-add), the first argument is for the attribute on the object that the error is associated with. There isn't a `:user` attribute though so that's not quite right.

That said, if you pass the symbol `:base` that indicates the error isn't associated with a specific attribute on the object. Therefor, I think you should replace `:user` here with `:base`